lib/src/common/util/crawler.dart

```diff
@@ -7,29 +7,19 @@ class RepositoryCrawler {
 
   RepositoryCrawler(this.github, this.slug);
 
-  Stream<GitHubFile> crawl() {
-    var controller = new StreamController<GitHubFile>();
-
-    var group = new FutureGroup();
-
-    void scan(String path) {
-      group.add(github.repositories.getContents(slug, path).then((contents) {
-        contents.tree.where((it) => it.type != "dir").forEach(controller.add);
-
-        contents.tree.where((it) {
-          return it.type == "dir";
-        }).forEach((file) {
-          scan(file.path);
-        });
-      }));
+  Stream<GitHubFile> crawl() async* {
+    Stream<GitHubFile> scan(String path) async* {
+      var contents = await github.repositories.getContents(slug, path);
+
+      for (var content in contents.tree) {
+        if (content.type == 'dir') {
```

Comment:
Just a nitpick, but I'd reverse the if so that it's written this way `if ('dir' == content.type) {`.

I feel it's just a better practice to avoid null pointer exceptions (just a random general advice).

Reply:
Forget about my comment, I'm crazy, sorry about that lol.